src/services/publications-api.js

```diff
@@ -0,0 +1,91 @@
+import axios from "axios"
+
+const apiClient = axios.create({
+    baseURL: 'http://localhost:3000/',
+    timeout: 1000
+})
+apiClient.interceptors.request.use(
+    (config)=>{
+        const userDetails = localStorage.getItem('user')
+        if(userDetails){
+            const token = JSON.parse(userDetails).token
+            config.headers.Authorization = `${token}`
+            console.log(token)
+        }
+        return config
+    },
+    (err)=> Promise.reject(err)
+)
+
+export const testConnection = async () => {
+    try {
+      // Realiza una solicitud GET de prueba al servidor backend
+      const response = await axios.get("http://localhost:3000/test");
+      
+      // Si la solicitud tiene éxito, devuelve true
+      return true;
+    } catch (error) {
+      // Si hay algún error, devuelve false
+      return false;
+    }
+  };
+
+// //////////////////////////////////////////////////// //
+// MÉTODOS PUBLICATIONS
+// //////////////////////////////////////////////////// //
+
+export const getPublications = async (searchQuery = "") => {
+    try {
+        const params = {};
+        if (searchQuery) params.q = searchQuery;
+
```

Comment:
no me gusta que `q` sea el parametro usemos nombres descriptivos siempre, no hay razon por la cual hacerlo asi

```suggestion
        if (searchQuery) params.query = searchQuery;
```

o

```suggestion
        if (searchQuery) params.search = searchQuery;

```